low-level/framework/utils/sysfs_interface.py

```diff
@@ -138,7 +140,144 @@ def fetch_nw_cable_status(self, nw_interface_path, interface):
                 carrier_indicator = cFile.read().strip()
             if carrier_indicator not in self.nw_phy_link_state.keys():
                 carrier_indicator = 'unknown'
-        except OSError as os_err:
-            return os_err.errno
+        except Exception:
+            carrier_indicator = 'unknown'
         return self.nw_phy_link_state[carrier_indicator]
 
+    def fetch_nw_operstate(self, interface):
+        """
+        Return the Operational Status of the Network Interface.
+
+        Possible Values: <'UNKNOWN', 'NOTPRESENT", 'DOWN', 'LOWERLAYERDOWN',
+                          'TESTING', 'DORMANT', 'UP'>
+        """
+        operstate_file_path = os.path.join(self.get_sys_dir_path('net'),
+                                           interface, 'operstate')
+        operstate = "UNKNOWN"
+        try:
+            if os.path.isfile(operstate_file_path):
+                with open(operstate_file_path) as cFile:
+                    operstate = cFile.read().strip().upper()
+        except Exception as err:
+            raise err
+        return operstate
+
+    def get_sas_host_list(self):
+        """
+            Return SAS host list from /sys/class/sas_host directory.
+
+            eg: ['host1', 'host2']
+        """
+        sas_hosts = []
+        try:
+            host_dirpath = self.get_sys_dir_path('sas_host')
+            sas_hosts = os.listdir(host_dirpath)
+            sas_hosts = [host for host in sas_hosts
+                         if 'host' in host]
+        except Exception:
+            return []
```

Comment:
reset sas_hosts = [], so one return statement @ line 179 would suffice

Reply:
updated in new PR.